src/app/public/modules/lookup/lookup-show-more-modal.component.ts

```diff
@@ -0,0 +1,331 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  SkyDataManagerState,
+  SkyDataViewConfig,
+  SkyDataManagerService
+} from '@skyux/data-manager';
+
+import {
+  SkyModalInstance
+} from '@skyux/modals';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  debounceTime,
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyLookupSelectMode
+} from './types/lookup-select-mode';
+
+import {
+  SkyLookupShowMoreContext
+} from './types/lookup-show-more-context';
+
+@Component({
+  selector: 'skyux-lookup-show-more-modal',
+  templateUrl: './lookup-show-more-modal.component.html',
+  styleUrls: ['./lookup-show-more-modal.component.scss'],
+  providers: [SkyDataManagerService],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyLookupShowMoreModalComponent implements OnDestroy, OnInit {
+
+  /**
+   * @internal
+   * Fires when users select the "Add" button
+   */
+  @Output()
+  public addClick: EventEmitter<void> = new EventEmitter();
+
+  public items: any[];
+
+  public dataManagerConfig = {
+    sortOptions: [
+      {
+        id: 'az',
+        label: 'Alphabetical (A - Z)',
+        descending: false,
+        propertyName: 'name'
+      },
+      {
+        id: 'za',
+        label: 'Alphabetical (Z - A)',
+        descending: true,
+        propertyName: 'name'
+      }
+    ]
+  };
+
+  public dataState = new SkyDataManagerState({});
+  public displayedItems: any[] = [];
+  public itemsHaveMore: boolean = false;
+  public viewId = 'repeaterView';
+  public viewConfig: SkyDataViewConfig = {
+    id: this.viewId,
+    name: 'Repeater View',
+    icon: 'list',
+    searchEnabled: true,
+    filterButtonEnabled: false,
+    multiselectToolbarEnabled: this.context.selectMode === 'multiple',
+    onClearAllClick: this.clearAll.bind(this),
+    onSelectAllClick: this.selectAll.bind(this)
+  };
+
+  private itemIndex: number = 0;
+  private ngUnsubscribe = new Subject();
+
+  constructor(
+    public modalInstance: SkyModalInstance,
+    public context: SkyLookupShowMoreContext,
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerService: SkyDataManagerService
+  ) { }
+
+  public ngOnInit(): void {
+    this.dataState.searchText = this.context.initialSearch;
+    this.dataManagerService.initDataView(this.viewConfig);
+
+    this.dataManagerService.initDataManager(
+      {
+        activeViewId: 'repeaterView',
+        dataManagerConfig: this.dataManagerConfig,
+        defaultDataState: this.dataState
+      }
+    );
+
+    setTimeout(() => {
```

Comment:
Mind adding a comment above this `setTimeout` explaining why it's needed?

Reply:
I actually dug into this and was able to get around it by moving this logic from `ngOnInit` to `ngAfterViewInit`. This is done in the `lookup-modals` PR